experimental/PropertyDDS/packages/property-properties/src/propertyFactory.js

```diff
@@ -1444,37 +1828,52 @@ class PropertyFactory {
+                        propertyDef.optionalChildren[id] = typeid;
                     }
 
                     if (valueParsed.value) {
-                        const property = this._createFromPropertyDeclaration(properties[i], undefined, in_scope, false);
-                        this._setInitialValue(property, valueParsed);
-                        if (optional) {
-                            in_parent._insert(property.getId(), property, true);
-                        } else {
-                            in_parent._append(property, in_allowChildMerges);
-                        }
+                        const newChildEntry = {
+                            initialValue: valueParsed,
+                            optional,
+                            allowChildMerges: in_allowChildMerges
+                        };
+                        propertyDef.children = propertyDef.children || [];
+                        propertyDef.children.unshift([properties[i].id, newChildEntry]);
+                        const property = this._createDefFromPropertyDeclaration(properties[i],
+                                                                                in_scope, newChildEntry);
                     } else if (!optional) {
-                        const property = this._createFromPropertyDeclaration(properties[i], undefined, in_scope, false);
-                        in_parent._append(property, in_allowChildMerges);
+                        const newChildEntry = {
+                            initialValue: undefined
+                        };
+                        propertyDef.children = propertyDef.children || []
+                        propertyDef.children.unshift([properties[i].id, newChildEntry]);
+                        const property = this._createDefFromPropertyDeclaration(properties[i],
+                                                                                in_scope, newChildEntry);
+
                     }
                 }
             }
 
-            if (in_evaluateConstants && in_template.constants) {
+            if (in_template.constants) {
                 const constants = in_template.constants;
                 for (let i = 0; i < constants.length; i++) {
                     const context = constants[i].context || 'single';
                     const valueParsed = this._parseTypedValue(constants[i], in_scope, context);
-                    const constant = this._createFromPropertyDeclaration(constants[i], undefined, in_scope, true);
+
+                    const newChildEntry = {
+                        initialValue: undefined,
+                        constant: true
+                    };
+                    propertyDef.children = propertyDef.children || [];
+                    propertyDef.children.unshift([constants[i].id, newChildEntry]);
+
+                    this._createDefFromPropertyDeclaration(constants[i], in_scope, newChildEntry);
```

Comment:
Function is a bit confusing. Sometimes it returns a result and sometimes relies on a side effect on one of the arguments.

Reply:
The function actually never returns anything any longer, but writes into the output parameter. The cases, where a return value was assigned, this was no used afterwards.